controllers/openstackdataplanerole_controller.go

```diff
@@ -267,7 +267,10 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 			return ctrl.Result{}, err
 		}
 	} else {
+		// We don't require any baremetal nodes, so we will remove these conditions from the status
 		instance.Status.Conditions.Remove(dataplanev1.RoleBareMetalProvisionReadyCondition)
+		instance.Status.Conditions.Remove(dataplanev1.RoleDNSDataReadyCondition)
```

Comment:
This is wrong. IPReservation and Dns is used for both pre-provisoned and baremetal deployments.